### PR TITLE
Make LanguageClientOptions.synchronize.fileEvents configurable  resolves #124

### LINKED ISSUE #124: Custom pomfile location
### Motivation
Working with teamcity kotlin dsl in vscode without making `.temcity/` directory as workspace root

### Description
Current file watcher https://github.com/fwcd/vscode-kotlin/blob/main/src/languageSetup.ts#L185 can't find pom.xml in `${workspaceRoot}/.teamcity/pom.xml` 'cause of it's glob pattern. So let's make pom.xml file and *.gradle files paths customizable

### Alternatives considered
I don't have any :c

### PR DESCRIPTION
**Pull request by tony-sol**
https://github.com/fwcd/vscode-kotlin/issues/124 
Make LanguageClientOptions.synchronize.fileEvents configurable via list of glob patterns as `kotlin.watchFiles` in settings.json

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -318,6 +318,18 @@
...
+                    "type": "array",
+                    "default": [
+                        "**/*.kt",
+                        "**/*.kts",
+                        "**/*.java",
+                        "**/pom.xml",
+                        "**/build.gradle",
+                        "**/settings.gradle"
+                    ],
+                    "description": "Specifies glob patterns of files, which would be watched by LSP client. The LSP client doesn't support watching files outside a workspace folder."
```
Comment (fwcd) [inline package.json:332]: For consistency, prefer the term "language server/client" over "LSP server/client":

```suggestion
                    "description": "Specifies glob patterns of files to be watched by the language client. The language client doesn't support watching files outside a workspace folder."
```

File: src/languageSetup.ts [hunk 2]
```diff
@@ -18,14 +18,14 @@ import { MainClassRequest, OverrideMemberRequest } from "./lspExtensions";
 export async function activateLanguageServer({ context, status, config, javaInstallation, javaOpts }: ServerSetupParams): Promise<KotlinApi> {
     LOG.info('Activating Kotlin Language Server...');
     status.update("Activating Kotlin Language Server...");
-    
+
```
Comment (fwcd) [inline src/languageSetup.ts:21]: I think it's fine here, but in general, please try to avoid including unrelated formatting changes in commits.

File: package.json [hunk 3]
```diff
@@ -318,6 +318,18 @@
                     "default": true,
                     "description": "[DEPRECATED] Specifies whether code completion should provide snippets (true) or plain-text items (false).",
                     "deprecationMessage": "Use 'kotlin.completion.snippets.enabled'"
+                },
+                "kotlin.watchFiles": {
```
Comment (fwcd) [inline package.json:322]: I'm a bit torn on whether we should name this `kotlin.watchFiles` or `kotlin.languageServer.watchFiles`. On the one hand, it doesn't related _directly_ to the execution of the language server, on the other hand, all "meta options" relating to the language server are prefixed with `kotlin.languageServer` (e.g. the port, transport, ...). The options under `kotlin` are generally intended for handling by the language server directly (this might be a questionable convention, but we should stay consistent where possible).

File: src/languageSetup.ts (hunk 2 above)
Comment (Tony-Sol) [reply src/languageSetup.ts:21]: Yeah, that was auto-remove of trailing whitespaces in ide. Sorry about that.

Should i revert this or let it pass?

**Comment by fwcd** [comment]
Thanks!
